src/test/java/seedu/address/model/AddressBookTest.java

```diff
@@ -76,6 +77,39 @@ public void hasTag_tagWithSameIdentityFieldsInAddressBook_returnsTrue() {
         assertTrue(addressBook.hasTag(editedCS2103));
     }
 
+
+    @Test
+    public void setTag_tagInAddressBook_returnsTrue() {
+        AddressBook dummyAddressBook = addressBook;
+        dummyAddressBook.addTag(CS2103);
+        dummyAddressBook.setTag(CS2103, CS2101);
+        Tag expectedTag = new TagBuilder(CS2101).build();
+        assertTrue(dummyAddressBook.hasTag(expectedTag));
+    }
+
+    @Test
+    public void setTag_nullTag_returnsNullPointerException() {
```

Comment:
Should the method name be `removeTag_nullTag_throwsNullPointerException`?

Reply:
Yes